api/iteration.go

```diff
@@ -48,15 +96,19 @@ func NewIteration(dir string, filenames []string) (*Iteration, error) {
 	// All the files should be within the exercism path.
 	for _, filename := range filenames {
 		if !iter.isValidFilepath(filename) {
-			return nil, errUnidentifiable
+			// User has run exercism submit in the wrong directory
+			return nil, newIterationError(msgSubmitCalledFromWrongDir, iter.Dir)
```

Comment:
I don't think we're getting any value from a custom type here. All we're doing is returning an error string, right?